### ISSUE TITLE
no join new user more than 2

### ISSUE THREAD

**Issue by Hadi14** [borjanebbal/webrtc-node-app#22]
hi
i want use this for more than 2 user. there is possible?

**Comment by borjanebbal** [comment]
Yes, WebRTC supports more than two users, but it requires a signalling and relay mechanism to manage connections efficiently. You can search for information about mesh, MCU and SFU architectures.

The code in this repository is structured for a 2-user WebRTC connection per room, where the second user joining triggers the exchange of WebRTC offer/answer messages. To enable a multi-user video call, some changes would be needed. It's not scalable, but a mesh architecture (P2P) can work for a small amount of users following this guide.

- Backend ([server.js](https://github.com/borjanebbal/webrtc-node-app/blob/main/server.js))
  - Allow unlimited users per room.
  - Send `new_peer` event to all users when someone joins.
  - Route WebRTC offers/answers and ICE candidates to specific users.
  - Notify users when a peer disconnects.
- Frontend ([client.js](https://github.com/borjanebbal/webrtc-node-app/blob/main/public/client.js))
  - Maintain a peer connection for each user in a dictionary.
  - Send offers to all existing users when joining.
  - Dynamically add and remove video elements as peers join/leave.
  - Ensure ICE candidates are sent to the right peer connection.
